src/main/resources/category/java/remoting.xml

```diff
@@ -281,25 +281,17 @@ return HttpClientBuilder.create() // good, setConnectionManager called, pool con
             <property name="tags" value="jpinpoint-rule,correctness" type="String" description="classification"/>
             <property name="xpath">
                 <value><![CDATA[
-(: locally created http client builder with setConnectionManager and at least one of setMaxConnTotal/setMaxConnPerRoute :)
-//MethodDeclaration//PrimaryExpression/PrimaryPrefix/Name[
-(
-(pmd-java:typeIs('org.apache.http.impl.client.HttpClientBuilder') and @Image='HttpClientBuilder.create')
-or (pmd-java:typeIs('org.apache.http.impl.nio.client.HttpAsyncClientBuilder') and @Image='HttpAsyncClientBuilder.create')
-or (pmd-java:typeIs('org.apache.http.impl.client.HttpClients') and @Image='HttpClients.custom')
-)
-and ancestor::MethodDeclaration//(PrimaryPrefix/Name|PrimarySuffix)[ends-with(@Image, 'setConnectionManager')]
-]/ancestor::MethodDeclaration//(PrimaryPrefix/Name|PrimarySuffix)[ends-with(@Image, 'setMaxConnTotal') or ends-with(@Image, 'setMaxConnPerRoute')]
-,
-(: method param http client builder setConnectionManager and at least one of setMaxConnTotal/setMaxConnPerRoute :)
-//MethodDeclaration//FormalParameter//ClassOrInterfaceType[
-(
-pmd-java:typeIs('org.apache.http.impl.client.HttpClient')
-or pmd-java:typeIs('org.apache.http.impl.nio.client.HttpAsyncClientBuilder')
-or pmd-java:typeIs('org.apache.http.impl.client.HttpClients')
-)
-and (ancestor::MethodDeclaration//(PrimaryPrefix/Name|PrimarySuffix)[ends-with(@Image, 'setConnectionManager')])
-]/ancestor::MethodDeclaration//(PrimaryPrefix/Name|PrimarySuffix)[ends-with(@Image, 'setMaxConnTotal') or ends-with(@Image, 'setMaxConnPerRoute')]
+//MethodDeclaration[
+    .//MethodCall/(TypeExpression|VariableAccess)[
+        pmd-java:typeIs('org.apache.http.impl.client.HttpClientBuilder')
+        or pmd-java:typeIs('org.apache.http.impl.client.HttpClients')
+        or pmd-java:typeIs('org.apache.http.impl.nio.client.HttpAsyncClientBuilder')
+    ]
+][
+    .//MethodCall[@MethodName='setConnectionManager']
+]
+(: trying to get to the right line, unfortunately, the chaining seems to make it not work :)
```

Comment:
hmm, is this always an issue with chaining? should we create an issue in pmd project for this?

Reply:
it pannes in designer as well as in the unit tests, so I guess also in IntellJ. Maybe wait until we checked with the newest pmd (7.6?)